File: verification/verify_app.py
from playwright.sync_api import sync_playwright, expect
import os

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={'width': 1280, 'height': 720})
        page = context.new_page()

        # Load the local HTML file
        file_path = os.path.abspath("index.html")
        page.goto(f"file://{file_path}")

        # Verify page title
        expect(page).to_have_title("GlassyTasks - Portfolio To-Do & Notes")

        # Verify Tabs exist
        expect(page.locator(".tab-btn").first).to_contain_text("Tasks")

        # --- To-Do Test ---
        # Add a task
        page.fill("#todo-input", "Buy groceries")
        page.click("#add-todo-btn")

        # Verify task added
        expect(page.locator(".task-text").first).to_have_text("Buy groceries")

        # Add a high priority task
        page.fill("#todo-input", "Finish project")
        page.select_option("#priority-select", "high")
        page.click("#add-todo-btn")

        # Verify order (newest on top)
        expect(page.locator(".task-text").first).to_have_text("Finish project")
        # Use to_contain_class instead of to_have_class for partial match
        expect(page.locator(".task-item").first).to_contain_class("task-item")

        # Complete the first task
        page.locator(".task-checkbox").first.click()

        # Switch to Active filter
        page.click("button[data-filter='active']")

        # Switch back to All
        page.click("button[data-filter='all']")

        # --- Notes Test ---
        # Switch to Notes tab
        page.click("button[data-tab='notes']")

        # Check empty state
        # Initially empty state is visible (if no notes) or hidden (if notes exist from local storage? new session should be empty)
        # But wait, localStorage persists across sessions in browser, but here we launch a fresh browser context?
        # Actually standard playwright launch creates incognito context usually, but localStorage might be empty.
        expect(page.locator("#empty-state-notes")).to_be_visible()

        # Create a note
        page.click("#add-note-btn")
        expect(page.locator("#note-modal")).to_be_visible()

        page.fill("#note-title-input", "Meeting Notes")
        page.fill("#note-body-input", "Discuss Q3 goals and roadmap.")
        page.click("#save-note-btn")

        # Verify note added
        expect(page.locator(".note-title").first).to_have_text("Meeting Notes")
        expect(page.locator("#empty-state-notes")).not_to_be_visible()

        # Take screenshot
        page.screenshot(path="verification/app_verification.png", full_page=True)

        browser.close()

if __name__ == "__main__":
    run()

File: verification/verify_app.js
const { chromium } = require('playwright');
const assert = require('assert');

(async () => {
  console.log('Starting verification...');
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // 1. Open the app
  await page.goto('http://localhost:8080');
  const title = await page.title();
  assert.match(title, /GlassyTasks/);
  console.log('Title verified');

  // 2. Create Task
  await page.fill('#taskInput', 'Test Task 1');
  await page.selectOption('#priorityInput', 'high');
  await page.click('#addTaskBtn');

  // Check for task existence
  await page.waitForSelector('.task-text:has-text("Test Task 1")');
  console.log('Task created');

  // 3. Verify Priority UI
  const priorityDot = await page.locator('.task-item:has-text("Test Task 1") .p-high');
  assert.ok(await priorityDot.isVisible(), 'High priority dot visible');
  console.log('Priority UI verified');

  // 4. Persistence Test (part 1)
  await page.fill('#taskInput', 'Task to Persist');
  await page.click('#addTaskBtn');
  await page.waitForSelector('.task-text:has-text("Task to Persist")');

  // 5. Notes Logic
  await page.click('#tabNotes');
  await page.waitForSelector('#notesSection', { state: 'visible' });

  await page.click('#addNoteBtn');
  await page.waitForSelector('#noteModal', { state: 'visible' });

  await page.fill('#noteTitle', 'Persistent Note');
  await page.fill('#noteBody', 'Body text');
  await page.click('#saveNoteBtn');

  await page.waitForSelector('.note-title:has-text("Persistent Note")');
  console.log('Note created');

  // 6. Reload and Verify Persistence
  await page.reload();
  await page.waitForSelector('#taskInput'); // Default view

  const taskVisible = await page.isVisible('.task-text:has-text("Task to Persist")');
  assert.ok(taskVisible, 'Task persisted after reload');
  console.log('Task persistence verified');

  const notes = await page.evaluate(() => JSON.parse(localStorage.getItem('glassy_notes')));
  assert.ok(notes && notes.length > 0, 'Notes exist in storage');
  assert.strictEqual(notes[0].title, 'Persistent Note');
  console.log('Note persistence verified');

  // 7. Visual Verification (Screenshot)
  await page.click('#tabNotes');
  await page.waitForSelector('.note-card');

  // WAIT FOR ANIMATION
  await page.waitForTimeout(1000);

  await page.screenshot({ path: 'verification/final_screenshot.png', fullPage: true });
  console.log('Screenshot captured at verification/final_screenshot.png');

  await browser.close();
  console.log('All verifications passed!');
})();
